cabal-install/cabal-install-solver/src/Distribution/Solver/Types/OptionalStanza.hs

```diff
@@ -1,5 +1,6 @@
 {-# LANGUAGE DeriveGeneric #-}
 {-# LANGUAGE DeriveDataTypeable #-}
+{-# OPTIONS_GHC -fno-warn-incomplete-record-updates #-}
```

Comment:
Why?

Reply:
Because these are an incomplete record update

https://github.com/haskell/cabal/blob/e76b56dc7b2eba7696f85844b690daa99e3eca20/cabal-install/cabal-install-solver/src/Distribution/Solver/Types/OptionalStanza.hs#L30-L31

See

https://github.com/haskell/cabal/blob/e76b56dc7b2eba7696f85844b690daa99e3eca20/Cabal/src/Distribution/Types/ComponentRequestedSpec.hs#L66-L70

Reply:
Rather, fix those. Make setters `setTestsRequested` and `setBenchmarkRequested`.

Reply:
I'm not sure what the intended semantics of those should be, specifically, what should they do on a `OneComponentRequestedSpec`?

As a compromise, how about the latest?